### DISCUSSION TITLE
laser corexy

### DISCUSSION THREAD

**Discussion by distebia** [grblHAL/core#571] [General]
Hi, I would like to build a machine with corexy kinematics. As a board I use mks dlc32 (esp32). I know that I can generate the firmware and enable the corexy option. However, I would also like to understand what changes to make and compile the firmware directly. Also, do the grbl settings ($23-44-46) remain the same when switching from cartesian (x and y) to corexy?

**Comment by terjeio** [comment]
CoreXY is enabled in [config.h](https://github.com/grblHAL/core/blob/7725fab5b8901c666259e010675d3259b4c8491b/config.h#L156-L168).

> Also, do the grbl settings ($23-44-46) remain the same when switching from cartesian (x and y) to corexy?

X and Y has to be homed separately

**Reply by distebia** [reply]
#define COREXY On, and enable corexy...$23=3 for homing bottom left...$44=1 for X and $45=2 for Y? is this correct?

**Reply by terjeio** [reply]
Yes, $-settings can be changed at run-tim if not.
